python/tests/test_tables.py

```diff
@@ -2266,3 +2266,133 @@ def test_set_columns_not_implemented(self):
+    """
+
+    def get_msprime_example(self, T=100, N=100, n=10):
+        # we assume after the split the ts are completely independent
+        M = [[0, 0], [0, 0]]
+        population_configurations = [
+            msprime.PopulationConfiguration(sample_size=n),
+            msprime.PopulationConfiguration(sample_size=n)
+        ]
+        demographic_events = [
+            msprime.CensusEvent(time=T),
+            msprime.MassMigration(T, source=1, dest=0, proportion=1)
+        ]
+        ts = msprime.simulate(
+            Ne=N,
+            population_configurations=population_configurations,
+            demographic_events=demographic_events,
+            migration_matrix=M,
+            length=1e4,
+            recombination_rate=1e-4,
+            mutation_rate=1e-4)
+        return ts
+
+    def verify_graft_simplification(self, tables, tablesg, node_map):
+        # check that if we simplify tablesg = tables.graft(tables2) back to
+        # tables.samples, we get the same trees: after simplification
+        # everything but provenance should be the same
+        nodes = [
+            k for k, n in enumerate(tables.nodes) if n.flags & tskit.NODE_IS_SAMPLE
+        ]
+        nodesg = [node_map[n] for n in nodes]
+        # simplifying with filter_populations=False bc SLiM can start
+        # with id 1 instead of zero
+        tables.simplify(nodes, filter_populations=False)
+        tablesg.simplify(nodesg, filter_populations=False)
+        # TODO: a test for provenances
+        tables.provenances.clear()
+        tablesg.provenances.clear()
+        self.assertEqual(len(tables.nodes), len(tablesg.nodes))
+        # simplify does not put individuals in order
+        # but it does for nodes. that is why we check
+        # for equality of individuals as follows
+        for j in range(tables.nodes.num_rows):
+            na = tables.nodes[j]
+            nb = tablesg.nodes[j]
+            if na.individual != -1 or nb.individual != -1:
+                self.assertFalse((na.individual == -1) or (nb.individual == -1))
+                ia = tables.individuals[na.individual]
+                ib = tablesg.individuals[nb.individual]
+                self.assertEqual(ia.flags, ib.flags)
+                self.assertEqual(ia.metadata, ib.metadata)
+                self.assertTrue((ia.location == ib.location).all())
+            if na.population != -1 or nb.population != -1:
+                self.assertFalse((na.population == -1) or (nb.population == -1))
+                pa = tables.populations[na.population]
+                pb = tablesg.populations[nb.population]
+                self.assertEqual(pa, pb)
+            self.assertEqual(na.time, nb.time)
+            self.assertEqual(na.metadata, nb.metadata)
+            self.assertEqual(na.flags, nb.flags)
+        tables.individuals.clear()
+        tablesg.individuals.clear()
+        tables.populations.clear()
+        tablesg.populations.clear()
+        tables.nodes.clear()
```

Comment:
Why clear the node tables? These should be equal, shouldn't they?

Reply:
We test the nodes above, but the tables are not exactly equal if because of the remapping of populations.